### ISSUE TITLE
High memory consumption

### ISSUE THREAD

**Issue by paulmenzel** [hexorx/countries#284]
Using [derailed_benchmarks](https://github.com/schneems/derailed_benchmarks) to benchmark a Ruby on Rails app requiring phony_rails, that depends on Countries 0.11.4, it shows that it uses around 18.7 MiB of memory.

```
TOP: 144.9038 MiB
[…]
  phony_rails: 22.7852 MiB
    iso3166: 18.7734 MiB
      countries/country: 18.7734 MiB
        i18n_data/file_data_provider: 0.0156 MiB
          fileutils: 0.0 MiB (Also required by: rails/application, rack/utils, and 4 others)
      i18n_data: 0.0 MiB
        i18n_data/version: 0.0 MiB
      iso4217: 0.0 MiB
        currencies/extentions: 0.0 MiB
        currencies/currency: 0.0 MiB
        currencies/exchange_bank: 0.0 MiB
          thread: 0.0 MiB (Also required by: active_support/dependencies, action_view/helpers/asset_tag_helpers/asset_paths, and 15 others)
      yaml: 0.0 MiB (Also required by: active_support/ordered_hash, active_support/core_ext/big_decimal/conversions, and 5 others)
    phony: 1.4805 MiB
[…]
```

There is the [pull request #1 for iso3166](https://github.com/vandrijevik/iso3166/pull/1) by @JuanitoFatas and [issue #107 in country_select](https://github.com/stefanpenner/country_select/issues/107) by @zywx.

**Comment by JuanitoFatas** [comment]
> There is the pull request #1 for iso3166 by @JuanitoFatas
This is not correct. The iso3166 which should be fixed is this one: https://github.com/hexorx/countries/blob/master/lib/iso3166.rb.

**Comment by rposborne** [comment]
Hey @paulmenzel thanks for the continued concern, I am actively working on a solution that balances both speed and memory usage. I plan for it to be our 1.0 release as it will break things :)

**Comment by rposborne** [comment]
Just pushed a memory refactor branch.  Below is a ```derailed bundle:mem``` output for a blank rails test app, with countries and the country select gem, also cut the spec suite down to less than 0.2 seconds woot

There is a pull request here https://github.com/hexorx/countries/pull/285 if you want to check it out

```
  countries: 0.4492 MiB
    iso3166: 0.4453 MiB (Also required by: country_select)
      iso4217: 0.3828 MiB
        currencies/currency: 0.3438 MiB
  uglifier: 0.3125 MiB
```

**Comment by rposborne** [comment]
I have merged this into master and released 1.0.0,  there are a few api changes around ```all``` so make sure to test your code first.
